build/parsePrep.c

```diff
@@ -450,6 +448,20 @@ int parsePrep(rpmSpec spec)
     }
 
 exit:
+    /* Issue a warning for deprecated %patchN usage */
+    int num = spec->numConverted;
+    if (num) {
+	const char *msg =
+	    _("%%patchN is deprecated, use %%patch N (or %%patch -P N)\n");
+	rpmlog(RPMLOG_WARNING, msg);
+	/* Issue one additional warning for any remaining conversions */
+	if (num == 2)
+	    rpmlog(RPMLOG_WARNING, msg);
+	if (num > 2)
+	    rpmlog(RPMLOG_WARNING,
+		   _("--- Last warning suppressed %i times ---\n"), num - 1);
```

Comment:
Oh, I would've just condensed it into one "patchN deprecated (repeated X times)" message. This is pretty fancy handling for just one message :smile:

Reply:
OK, true :smile: This is basically the format of the general rpmlog-native solution that I've started separately, but for now (or for this particular message), it's indeed too elaborate. I'll push a simplified version later today/tomorrow. Thanks!